lib/kml/networklink.ts

```diff
@@ -0,0 +1,232 @@
+import type { Feature, Polygon } from "geojson";
+import type { KMLOptions } from "lib/kml";
+import { type StyleMap, get1, getMulti, num1 } from "../shared";
+import { extractIconHref, extractStyle } from "./extractStyle";
+import {
+    type Schema,
+    extractCascadedStyle,
+    extractExtendedData,
+    extractTimeSpan,
+    extractTimeStamp,
+    getMaybeHTMLDescription,
+} from "./shared";
+
+interface BoxGeometry {
+    bbox?: BBox;
+    geometry: Polygon;
+}
+
+enum AltitudeMode {
+    ABSOLUTE = "absolute",
+    RELATIVE_TO_GROUND = "relativeToGround",
+    CLAMP_TO_GROUND = "clampToGround",
+    CLAMP_TO_SEAFLOOR = "clampToSeaFloor",
+    RELATIVE_TO_SEAFLOOR = "relativeToSeaFloor",
+}
+
+type BBox = [number, number, number, number];
```

Comment:
There's also a `type BBox` in lib/kml/ground_overlay.ts, could we move this to lib/kml/shared to avoid repeating?